### DISCUSSION TITLE
Changing a crouch toggle to a hold-to-crouch...

### DISCUSSION THREAD

**Discussion by Knuckle-Bucket** [AntiMicroX/antimicrox#906] [Q&A]
Changing a hold to a toggle is simple, but it is proving quite difficult to change a toggle to a hold. 

In game, pressing [key] crouches, and pressing [key] a second time stands back up. I am trying to get it so that holding [key] down crouches, and releasing [key] stands back up.  
I've gotten this to work natively on the keyboard using autohotkey, but when I try to send the input from a gamepad with AntimicroX (or JoytoKey just to troubleshoot), it no longer functions properly.  I also couldn't get to work using the release function within AntiMicroX.  

 If it helps -- this is the autohotkey script that gives the correct functionality on the keyboard natively, using the K key:
 
 *K::
    Send {K Down}
    Sleep 50
    Send {K Up}
    KeyWait K
    Send {K Down}
    Sleep 50
    Send {K Up}
 
 I have tried the same sequence applied with releases and delays in AntiMicroX, but I don't know if I don't have a correct understanding of how it works within the program, or if it is just not working for some other reason. Is there some basic thing I am missing about how to achieve the function I am looking for? I don't care if it is a way to make the AHK script work with the gamepad, or a way to achieve it through AntiMicroX. I just want the function.  Any ideas?

**Comment by Knuckle-Bucket** [comment]
Turns out I was making this much more complicated than I needed to...  I ended up just using Steam reamapping, and it works with no trouble.

**Comment by MrChillings** [comment]
for those wondering how to do this you can achive this by making an advanced assignment like this:

![afbeelding](https://github.com/AntiMicroX/antimicrox/assets/154590517/1518560d-244a-4ecf-a5c4-3fa3a786242a)

first you assign the first key press "K" (now you are crouching while holding the button)  --> second you assign a release( when held for ... ms the key assigned after release will be triggered)  --> Then assign a key you want to perform when releasing the button (in this instance "K" again to stand up)

**Reply by Knuckle-Bucket** [reply]
I appreciate the response. What you show here is what I had initially tried, but there must be some other factor preventing it from working. The same instruction works with Steam remapping, and with the ReWASD, but there must be something different about how they communicate with the game/program in question (RPCS3 - PS3 emulator) compared to AntiMicro and AHK.  This sends the input I would expect when I test it in a notepad, but there is something about this particular thing that just doesn't want to work.  AntiMicro has worked with everything else I have tried it on so far, and when I can use it, I greatly prefer its much more user-friendly design; but it just doesn't seem like an option for this, unfortunately.

**Reply by MrChillings** [reply]
you have tried to run the program as administrator?

**Reply by Knuckle-Bucket** [reply]
Yes, I did.

**Comment by BoomerangAide** [comment]
Have you tried with a release time of 0?
Otherwise, there may be some possible trick involving "Turbo" but I cant get past "I think something like that could work" to actually try something. Maybe someone else will be inspired?
